kimchi/js/bindings.js

```diff
@@ -1219,6 +1219,32 @@ var caml_fq_plonk_gate_to_rust = function (gate) {
   );
 };
 
+// Provides: caml_fp_lookup_table_to_rust
+// Requires: plonk_wasm, caml_fp_vector_of_rust
+var caml_fp_lookup_table_to_rust = function (caml_lookup_table, mk_class) {
+  // removing 1 for the 0 used by jsoo to represent values.
+  var lt_len = caml_lookup_table.data.length - 1;
```

Comment:
`js_of_ocaml` represents OCaml records as arrays. You'll need to fix this to handle that style instead.
I'm also suspicious of the `i32` conversion; you should check that jsoo does indeed handle `i32` as a plain number.

Reply:
The changes are done. i32 seems to be number, see [readme](https://github.com/ocsigen/js_of_ocaml/#data-representation).